### PR TITLE
chore: upgrade to @types/node@10.x

### PR DESCRIPTION
**Pull request by raymondfeng**
Upgrade to `@types/node@10.x`

## Checklist

- [x] `npm test` passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] API Documentation in code was updated
- [ ] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [x] Affected artifact templates in `packages/cli` were updated
- [x] Affected example projects in `examples/*` were updated

### CODE REVIEW

File: packages/rest/src/rest.server.ts [hunk 1]
```diff
@@ -585,7 +586,8 @@ export class RestServer extends Context implements Server, HttpServerLike {
 
     return new Promise<void>((resolve, reject) => {
       httpServer.once('listening', () => {
-        this.bind(RestBindings.PORT).to(httpServer.address().port);
+        const address = httpServer.address() as AddressInfo;
+        this.bind(RestBindings.PORT).to(address.port);
```
Comment (shimks) [inline packages/rest/src/rest.server.ts]: Is node 10 causing a compilation error here?

Comment (bajtos) [reply packages/rest/src/rest.server.ts]: Yeah, this change is suspicious. 

@raymondfeng What is the return type of `httpServer.address()`? Why is an explicit cast to `AddressInfo` needed?

Comment (raymondfeng) [reply packages/rest/src/rest.server.ts]: See https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node/index.d.ts#L2725

Comment (bajtos) [reply packages/rest/src/rest.server.ts]: Weird, isn't it?

Do we know under which conditions we can receive `string` result? Do we need to handle this string case here, I mean is it safe to assume that we will always receive AddressInfo?

Not a big deal, we can investigate that later.

Comment (raymondfeng) [reply packages/rest/src/rest.server.ts]: FYI - For a server listening on a pipe or UNIX domain socket, the name is returned as a string. We should be safe here.

**Pull request merged**